lusidtools/cocoon/utilities.py

```diff
@@ -260,27 +260,40 @@ def set_attributes_recursive(
     # For each of the attributes to populate
     for key in list(populate_attributes):
 
+        # Get the attribute type
+        attribute_type = obj_attr[key]
+
         # If it is an additional attribute, populate it with the provided values and move to the next attribute
         if key in list(additional_attributes.keys()):
-            obj_init_values[key] = additional_attributes[key]
-            continue
+            # Handle identifiers provided within instrument definition (e.g. 'Bond', 'Future', etc.)
+            if (key, attribute_type) == ("identifiers", "dict(str, str)"):
+                obj_init_values[key] = {
+                    str_key: row[str_value["value"]]
+                    for str_key, str_value in mapping[key].items()
```

Comment:
Do we need to add `if not pd.isna(row[mapping[key]])` to ensure that we don't attempt to get a value from the DataFrame when it is `NaN` or `None`?

Reply:
Yes you are right. In fact something to add a test for, and I think we'll want to ensure it opens an empty dict() when it's NaN. Otherwise, one is required to map the identifier a second time, which is a mistake that people will surely make given the variable is set as nullable.

Reply:
Yes, great point about a dictionary if all are NaN! I wonder if that would be the behaviour anyway if all are NaN or not? Like you say best way to see is via a test!